File: python/caesar_cipher.py
class CesarCipher(object):
    _alphabet = None
    _shift = None

    def set_alphabet(self, value):
        self._alphabet = value

    def set_shift(self, value):
        self._shift = value

    def encode(self, plainText):
        encoded_text = ''
        alphabet = self._alphabet
        shift = self._shift

        for i in range(0, len(plainText)):
            if plainText[i] in alphabet:
                if alphabet.index(alphabet[alphabet.index(plainText[i])]) + shift >= len(alphabet):
                    shifted_index = alphabet.index(plainText[i]) + shift

                    while shifted_index >= len(alphabet):
                        shifted_index -= len(alphabet)

                    encoded_text += alphabet[shifted_index]

                else:
                    encoded_text += alphabet[alphabet.index(plainText[i]) + shift]
            else:
                encoded_text += plainText[i]
            
        return encoded_text

File: ruby/caesar_cipher.rb
class CesarCipher
  attr_writer :alphabet, :shift

  def initialize(alphabet, shift)
    @alphabet = alphabet
    @shift = shift  
  end

  def encode(plain_text)
    if @alphabet.kind_of?(Array)
      @alphabet = @alphabet.join("")
    end
    
    encoded_text = ''

    plain_text.size.times do |i|
        if @alphabet[plain_text[i]] != nil
            if @alphabet.index(plain_text[i]) + @shift >= @alphabet.size
                shifted_index = @alphabet.index(plain_text[i]) + @shift - @alphabet.size
                while shifted_index >= @alphabet.size
                    shifted_index -= @alphabet.size
                end
                encoded_text += @alphabet[shifted_index]
            else
                encoded_text += @alphabet[@alphabet.index(plain_text[i]) + @shift]
            end
        else
            encoded_text += plain_text[i]
        end
    end
        
    return encoded_text
  end

end 

